microprofile-health/src/main/java/org/jboss/eap/qe/microprofile/health/integration/CustomConfigSource.java

```diff
@@ -0,0 +1,43 @@
+package org.jboss.eap.qe.microprofile.health.integration;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+public class CustomConfigSource implements ConfigSource {
+    public static final String FILEPATH_PROPERTY = "config.source.properties.path";
+
+    @Override
+    public Map<String, String> getProperties() {
+        String filename = System.getProperty(FILEPATH_PROPERTY);
+        if (filename == null) {
+            throw new RuntimeException(FILEPATH_PROPERTY + " property not defined");
+        }
+        Map<String, String> props = new HashMap<>();
+        try (FileInputStream is = new FileInputStream(filename)) {
+            Properties properties = new Properties();
+            properties.load(is);
+            for (String key : properties.stringPropertyNames()) {
+                props.put(key, properties.getProperty(key));
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Do you really want to just print the stack trace and continue?

Reply:
Good point. I'll throw an `RuntimeException`